Automattic/vip-go-mu-plugins
# (CANTINA-949) Jetpack: Move pre_option_jetpack_active_plan around to run before JP loads

## Description
Fixes the issue of modules being deactivated by `check_plan_deactivate_modules()`. Also, removes `Jetpack::is_active()` check in the option filtering since it calls `get_userdata()` which isn't available at that point of code execution.

## Changelog Description

### Plugin Updated: VIP Jetpack

Fixes the issue of modules being deactivated by `check_plan_deactivate_modules()`.

## Pre-review checklist

Please make sure the items below have been covered before requesting a review:


## Pre-deploy checklist


## Steps to Test
To reproduce issue:
1) `wp jetpack module activate google-analytics`
2) `wp jetpack module list` and verify `google-analytics` module is activated
3) Go to wp-admin > Jetpack > Dashboard and then run step 2) and see that it gets deactivated.

This patch should fix that.
